test/verify/check-machines

```diff
@@ -3435,6 +3438,73 @@ class TestMachines(MachineCase, StorageHelpers, NetworkHelpers):
         b.click('#deactivate-network-test_network4-{0}'.format(connectionName)) # Deactivate transient network
         self.waitNetworkRow("test_network4", connectionName, False) # Check it's not present after deactivation
 
+    def testSnapshots(self):
+        # Checks if difference of @time1 and @time2 is not greater than @difference (in seconds)
+        def checkTimeDiff(time1, time2, difference):
+            tmp = time2.split(' ') # split "Today 13:13:13" into day and time
+            if not tmp[1].startswith('00:') and not tmp[0] == "Today":
+                return False
+
+            diff = datetime.strptime(time1, '%H:%M:%S') - datetime.strptime(tmp[1] + tmp[2], '%I:%M%p')
+            return -difference < diff.total_seconds() < difference
+
+        b = self.browser
+        m = self.machine
+
+        self.startVm("subVmTest1")
+
+        if m.image in distrosWithoutSnapshot:
+            return
+
+        self.login_and_go("/machines")
+        b.wait_in_text("body", "Virtual Machines")
+        self.waitVmRow("subVmTest1")
+        self.toggleVmRow("subVmTest1")
+        b.wait_in_text("#vm-subVmTest1-state", "running")
+        b.click("#vm-subVmTest1-snapshots") # open the "Disks" subtab
+        b.wait_in_text(".no-snapshots", "No snapshots")
+
+        # Check snapshot for running VM
+        m.execute("virsh detach-disk --domain subVmTest1 --target vda --persistent") # vda is raw disk, which are not supported by internal snapshots
+        m.execute("qemu-img create -f qcow2 /var/lib/libvirt/images/foobar.qcow2 1M")
+        m.execute("virsh attach-disk --domain subVmTest1 --source /var/lib/libvirt/images/foobar.qcow2 --target vdb --persistent")
+        time1 = datetime.now().strftime("%H:%M:%S")
+        m.execute("virsh snapshot-create-as --domain subVmTest1 --name snapshot1 --description 'Description of snapshot1' --disk-only")
+
+        b.reload() # snapshots events not available yet: https://gitlab.com/libvirt/libvirt/-/issues/44
+        b.enter_page('/machines')
+        b.wait_in_text("body", "Virtual Machines")
+        self.toggleVmRow("subVmTest1")
+        b.click("#vm-subVmTest1-snapshots")
+
+        b.wait_in_text("#vm-subVmTest1-snapshot-0-name", "snapshot1")
+        b.wait_in_text("#vm-subVmTest1-snapshot-0-description", "Description of snapshot1")
+        b.wait_in_text("#vm-subVmTest1-snapshot-0-type", "disk-snapshot")
```

Comment:
The previous conversation was resolved without any change.

This is still confusing me. `const statesMap` maps these internal strings to human readable names, but here it apparently still renders the original key instead of the mapped value?

Reply:
Oh right. The mapping is correct, it's just bad string in test. It's even failing the test.